zebrad/src/components/mempool.rs

```diff
@@ -141,6 +146,24 @@ impl Service<Request> for Mempool {
                 let _ = self.storage.insert(tx);
             }
         }
+
+        let mut storage = self.storage.clone();
+        let mut state = self.state.clone();
+
+        let _ = tokio::task::spawn(async move {
+            let tip_height = match state
+                .ready_and()
+                .await
+                .unwrap()
+                .call(zs::Request::Tip)
+                .await
+            {
+                Ok(zs::Response::Tip(Some((height, _hash)))) => height,
+                _ => panic!("A valid tip height is needed here"),
```

Comment:
Some of the existing tests will fail for this, i think they need to be fixed as having a tip at this point should be a must.

Reply:
With https://github.com/ZcashFoundation/zebra/pull/2774/commits/fc47b1403f75e8030e07d3e94c696af50f436f9c the panic is now gone as we ignore the code if we have no block tip from `best_tip_height()`. This looks correct to me however we can add an `else` arm if we need to handle this.

Reply:
I don't think we need to remove any transactions when the state is empty.
Can you double-check my reasoning here?

* transactions with expiry height `0` never expire, this is a consensus rule
* the lowest valid expiry height is `1`
* the lowest tip height that can expire transactions is `2`, because transactions expire when the tip is greater
* if the state is empty (`tip_height == None`), it hasn't reached height `2` yet, so there are no expired transactions

So this code looks correct. But let's test `remove_expired_transactions` with expiry and tip heights of `0`, `1`, and `2`, to prevent future bugs.